src_kolko_krzyżyk/TicTacToe_expand.py

```diff
@@ -185,35 +185,36 @@ def main(self):
             else:
                 player = 0
 
-def menu():
-    print('''\n\n\t"Simple game for two person in Tic Tac Toe.
-     Player decides how big the board is. Valid board (2 to n)
-     where board are matrix n x n. Good Luck and Have Fun."\n\n''')
-    # default matrix_choice
-    matrix_choice = 3
+def matrix_choice():
     while True:
         matrix_choice = input('Please enter size of martix from two to n: ')
         try:
             matrix_choice = int(matrix_choice)
-            break
+            return matrix_choice
         except:
             print("Dosen't entered number please try again" )
 
+def deletion_condition_to_win(matrix_size):
     while True:
         deletion_choice = input('Please enter number of deletions:'
                                         'Note: 1 <= deletions <= matrix: ')
         try:
             deletion_choice = int(deletion_choice)
-            if 1 <= deletion_choice <= matrix_choice:
-                return TicTacToe(matrix_choice, deletion_choice).main()
+            if 1 <= deletion_choice <= matrix_size:
+                return TicTacToe(matrix_size, deletion_choice).main()
             else:
                 print('Error deletions number')
         except:
             print("Dosen't entered number please try again")
 
+def menu():
+    print('''\n\n\t"Simple game for two person in Tic Tac Toe.
+     Player decides how big the board is. Valid board (2 to n)
+     where board are matrix n x n. Good Luck and Have Fun."\n\n''')
+    matrix_size = matrix_choice()
+    return deletion_condition_to_win(matrix_size)
```

Comment:
Zaufaj mi, to rozwiązanie jest lepsze. Obecnie funkcja deletion_condition_to_win miesza ze sobą odpalanie logiki gry z pytaniem użytkownika o warunek wygranej. Tego typu rozwiązania są mniej generyczne i przy zmianie interface'u (np. wyglądu gry) będziesz musiał przepisywać większość kodu na nowo. Negatywnych konsekwencji tego rozwiązanie jest jeszcze znacznie więcej.
```
wining_condition = get_wining_condition()
return TicTacToe(matrix_size, wining_condition).main()
```

Reply:
Rozumiem o co ci chodziło i masz racje.
A tak zapytam apropo tego kodu czy jesli mamy klasy a w klasach metody, to dobrym zwyczajem jest umieszczanie jeszcze funkcji po za klasa tak jak zrobiliśmy z funkcja menu wraz z matrix_size i deletion_condition ?? Czy po prostu stworzyc kolejna klase i wpakować to do stworzonej klasy tylko w tym przypadku to raczej nie ma sensu tak mi się wydaje. Lub też wpakować to do istniejącej klasy jako metody i odwołać się potem po instancji ??
I jeszcze jedna rzecz nie wiem czy zauważyłeś ale zostało wprowadzone więcej kodu lecz jak robiłem pull requesta te zmiany zostały juz scalone wcześniej...
Wiec twój komentarz odnośnie ze wysyłam w porcjach był miły ale wiem że dałem dupy ...
Bo zmian było dużo i robiłem to w jednym branchu... mam nauczkę ....